doobie-hikari/src/main/scala/com/avast/sst/doobie/DoobieHikariModule.scala

```diff
@@ -0,0 +1,46 @@
+package com.avast.sst.doobie
+
+import cats.effect.{Async, Blocker, ContextShift, Resource, Sync}
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.metrics.MetricsTrackerFactory
+import doobie.hikari.HikariTransactor
+
+import scala.concurrent.ExecutionContext
+
+object DoobieHikariModule {
+
+  /** Makes [[doobie.hikari.HikariTransactor]] initialized with the given config.
+    *
+    * @param boundedConnectExecutionContext [[scala.concurrent.ExecutionContext]] used for creating connections (should be bounded!)
+    */
+  def make[F[_]: Async](
+    config: DoobieHikariConfig,
+    boundedConnectExecutionContext: ExecutionContext,
+    blocker: Blocker,
+    metricsTrackerFactory: Option[MetricsTrackerFactory] = None
+  )(implicit cs: ContextShift[F]): Resource[F, HikariTransactor[F]] = {
+    for {
+      hikariConfig <- Resource.liftF {
+                       Sync[F].delay {
+                         val c = new HikariConfig()
```

Comment:
i would move this into it's own function that can be pure and not wrapped in `F`

Reply:
I moved it to its own function but since this is Java code I cannot believe it's not doing some side effects so I definitely need to wrap it in F. You can actually see it might try to load some properties from file so just to be sure I will wrap it.